rocketpool-cli/commands/wallet/rebuild.go

```diff
@@ -54,41 +54,52 @@ func rebuildWallet(c *cli.Context) error {
 		}(customKeyPasswordFile)
 	}
 
+	var enablePartialRebuildValue = false
+	if enablePartialRebuild.Name != "" {
+		enablePartialRebuildValue = c.Bool(enablePartialRebuild.Name)
+	}
+
 	// Log
 	fmt.Println("Rebuilding node validator keystores...")
+	fmt.Printf("Partial rebuild enabled: %s.\n", enablePartialRebuild.Name)
 
 	// Rebuild wallet
-	response, err := rp.Api.Wallet.Rebuild()
-	if err != nil {
-		return err
+	response, _ := rp.Api.Wallet.Rebuild(enablePartialRebuildValue)
+
+	// Handle and print failure reasons with associated public keys
+	if len(response.Data.FailureReasons) > 0 {
+		fmt.Println("Failure reasons:")
+		for pubkey, reason := range response.Data.FailureReasons {
+			fmt.Printf("Public Key: %s - Failure Reason: %s\n", pubkey.Hex(), reason)
+		}
+	} else {
+		fmt.Println("No failures reported.")
 	}
 
-	// Log & return
-	fmt.Println("The node wallet was successfully rebuilt.")
-	if len(response.Data.ValidatorKeys) > 0 {
+	fmt.Println("The response for rebuilding the node wallet was successfully received.")
+	if len(response.Data.RebuiltValidatorKeys) > 0 {
 		fmt.Println("Validator keys:")
-		for _, key := range response.Data.ValidatorKeys {
+		for _, key := range response.Data.RebuiltValidatorKeys {
 			fmt.Println(key.Hex())
```

Comment:
```suggestion
		for i, key := range response.Data.RebuiltValidatorKeys {
			fmt.Printf("\t%d.%s", i+1, key.Hex())
```

bit more ergonomic to enumerate the returned keys (starting with 1) so users can quickly see that the expected number of validators were recovered.